edb/server/mng_port/edgecon.pyx

```diff
@@ -812,8 +841,23 @@ cdef class EdgeConnection:
         cdef:
             WriteBuffer msg
             WriteBuffer packet
+            uint64_t allow_capabilities = ALL_CAPABILITIES
 
-        self.reject_headers()
+        headers = self.parse_headers()
+        if headers:
+            for k, v in headers.items():
+                if k == QUERY_OPT_ALLOW_CAPABILITIES:
+                    if len(v) != 8:
+                        raise errors.BinaryProtocolError(
+                            f'capabilities header must be exactly 8 bytes'
+                        )
+                    allow_capabilities = hton.unpack_uint64(
+                        cpython.PyBytes_AS_STRING(v))
+                    allow_capabilities |= ALWAYS_ALLOWED
```

Comment:
This code is repeated in a couple of places. Please factor it out into a method, the file is long enough already.

Reply:
Done. Thanks!